cardano-cli/src/Cardano/CLI/Byron/Run.hs

```diff
@@ -130,7 +131,10 @@ runMigrateDelegateKeyFrom
         -> ExceptT ByronClientCmdError IO ()
 runMigrateDelegateKeyFrom oldKeyformat oldKey (NewSigningKeyFile newKey) = do
   sk <- firstExceptT ByronCmdKeyFailure $ readByronSigningKey oldKeyformat oldKey
-  firstExceptT ByronCmdHelpersError . ensureNewFileLBS newKey $ serialiseByronWitness sk
+  migratedWitness <- case sk of
+                       LegacyWitness (ByronSigningKeyLegacy sKey) -> return . NonLegacyWitness $ ByronSigningKey sKey
+                       NonLegacyWitness _ -> panic "TODO: Can only migrate from legacy witness"
```

Comment:
We could fail a bit more gracefully here :smiley:

Reply:
Yep, this was an oversight.